anti_patterns/borrow_clone.md

```diff
@@ -0,0 +1,55 @@
+# Clone to satisfy the borrow checker
+
+## Description
+
+The borrow checker prevents Rust users from developing otherwise unsafe code by
+ensuring that either: only one mutable reference exists, or potentially many but
+all immutable references exist. If the code written does not hold true to these 
+conditions, this anti-pattern arises when the developer resolves the compiler
+error by cloning the variable.
+
+
+## Example
+
+```rust
+// define any variable
+let mut x = 5;
+
+// Borrow `x` -- but clone it first
+let y = &mut (x.clone()); 
+
+// perform some action on the borrow to prevent rust from optimizing this
+//out of existence
+*y += 1;
+
+// without the x.clone() two lines prior, this line would fail on compile as
+// x has been borrowed
+// thanks to x.clone(), x was never borrowed, and this line will run.
+println!("{}", x);
+```
+
+
+## Motivation
+
+It is tempting, particularly for beginners, to use this pattern to resolve
+confusing issues with the borrow checker. However, there are serious
+consequences. Using .clone() causes a copy of the data to be made. Any changes
+between the two are not synchronized -- as if two completely separate variables
+exist.
```

Comment:
I thought this is already very clear. `clone` word should be pretty clear. Just my comment, if people reach here I guess they should already know what clone does?

Reply:
I think it goes in line with the sentence before, that it is a problem "beginners" often run into when trying to satisfy the borrow checker. So I feel it's also ok to explain it once more to lower the barrier of entry to this topic, as many people from other programming languages struggle with it.